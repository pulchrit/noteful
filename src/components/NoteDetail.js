import React from 'react';
import PropTypes from 'prop-types';
import "../css/NoteDetail.css";

const NoteDetail = (props) => (
    <section>
        <p className="note-content">{props.content}</p>
    </section>
)

export default NoteDetail;

NoteDetail.propTypes = {
    content: PropTypes.string
};