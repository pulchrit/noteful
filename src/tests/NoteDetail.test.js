import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import NoteDetail from '../components/NoteDetail';

const content = "Veritatis porro minima perspiciatis. Repellat veniam quo iste ut. Iusto voluptas quae quibusdam. Odit neque iusto cupiditate iste quam. Fuga itaque aut praesentium ullam saepe ut et vero.\n \rQuisquam doloremque molestiae. Enim rerum dolorem et velit itaque magnam laborum. Aut officiis porro.\n \rQuae eum eaque error. Sed itaque ipsam nam provident aut voluptate. Perferendis repudiandae sequi laudantium est est animi eum. Unde alias et doloribus est hic et. Sed distinctio incidunt maiores aut voluptatibus et omnis mollitia fugit.";


describe("NoteDetail component", () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <NoteDetail />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the note content", () => {
      const wrapper = shallow(<BrowserRouter><NoteDetail content={content} /></BrowserRouter>);
      expect(toJson(wrapper)).toMatchSnapshot();
    })
});
