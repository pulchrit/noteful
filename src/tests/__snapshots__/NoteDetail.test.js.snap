// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteDetail component renders the note content 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <NoteDetail
    content="Veritatis porro minima perspiciatis. Repellat veniam quo iste ut. Iusto voluptas quae quibusdam. Odit neque iusto cupiditate iste quam. Fuga itaque aut praesentium ullam saepe ut et vero.
 
Quisquam doloremque molestiae. Enim rerum dolorem et velit itaque magnam laborum. Aut officiis porro.
 
Quae eum eaque error. Sed itaque ipsam nam provident aut voluptate. Perferendis repudiandae sequi laudantium est est animi eum. Unde alias et doloribus est hic et. Sed distinctio incidunt maiores aut voluptatibus et omnis mollitia fugit."
  />
</Router>
`;
